de-lib/src/main/java/org/iplantc/de/desktop/client/presenter/DesktopPresenterImpl.java

```diff
@@ -281,11 +284,18 @@ public void doLogout(boolean sessionTimeout) {
         }
     }
 
-    public void onBootstrapError(Integer statusCode) {
+    public void onBootstrapError(Integer statusCode, String errorMessage) {
         String redirectUrl = GWT.getHostPageBaseURL() + deClientConstants.errorUrl();
         if (statusCode != null) {
             redirectUrl += "-" + statusCode.toString();
         }
+        if (errorMessage != null) {
+            JSONObject json = jsonUtil.getObject(errorMessage);
+            String errorString = json.keySet().stream()
+                                     .map(key -> URL.encode(key) + "=" + URL.encode(json.get(key).toString()))
+                                     .collect(Collectors.joining("&"));
+            redirectUrl += "?" + URL.encode(errorString);
+        }
```

Comment:
✨

Reply:
That's a really good question. I was trying to maintain some feedback I got from Kathleen when we first put the error page together about how "ugly" and confusing it is to see error details and that they should be hidden unless the user wants to see them.  At least back then, it was kind of in line with the rest of the DE because we were leaning more heavily on friendly error messages in the `IplantAnnouncer` instead of displaying the raw error strings in `ErrorHandler`, but we've been utilizing the `ErrorHandler` more recently.  I could go either way - I like the cleaner look, but maybe it's more efficient to have the details right there?

Reply:
The reason why i asked this question is that we are recommending users to include the details when they contact the support. If we want to encourage users to include them, we should try to remove all additional steps (like clicking on the button to view details).

Reply:
On the other hand, if the error details are always showing, then I'd be worried that users would never read the last part of the error page, about checking our maintenance page, etc.

The information is segregated nicely with the error details inside the expansion panel. Maybe we could at least have the expansion panel expanded by default, so the user can collapse it if they want to only read the user-friendly messages below it?

Reply:
I am ok with having them expanded by default. Speaking of maintenance, we should link our public maintenance calendar in this page - https://www.cyverse.org/maintenance-calendar

Reply:
OK, should I reword the bottom blurb to mention the maintenance calendar separately or maybe leave it the same but make "...as well as `scheduled maintenance` events...` become a link to the calendar?

Reply:
I am ok with `scheduled maintenance` as link